src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -1450,6 +1450,11 @@ void MainWindow::slot_open_cornellbox_builtin_project()
     if (!can_close_project())
         return;
 
+    // Reset the ScenePickers for certain handlers, so that they don't hold a reference to
+    // Lazy<Tree> and consequently it can be destroyed without obstruction.
+    for (const_each<RenderTabCollection> i = m_render_tabs; i; ++i)
+        i->second->update_handlers();
```

Comment:
This snipped of code will make your code work only for the cornellbox project. There is a method `on_project_change` in this class, check it out. I also think all tabs are recreated when the project changed.

Reply:
Actually, loading of the built-in Cornell Box project seems like a _special_ case, where, we _first_ load the new project on the [line](https://github.com/appleseedhq/appleseed/blob/1a0907c5dc3c491686cb2be4b3d4f9410965ba35/src/appleseed.studio/mainwindow/mainwindow.cpp#L1458), and _then_ call `on_project_change` on the [line](https://github.com/appleseedhq/appleseed/blob/1a0907c5dc3c491686cb2be4b3d4f9410965ba35/src/appleseed.studio/mainwindow/mainwindow.cpp#L1461) which, as you said recreates all the `RenderTab`s,

Whereas, in other projects (loaded via the Open Project menu item in appleseed.studio), we _first_ remove the `RenderTab`s on the [line](https://github.com/appleseedhq/appleseed/blob/1a0907c5dc3c491686cb2be4b3d4f9410965ba35/src/appleseed.studio/mainwindow/mainwindow.cpp#L238) and _then_ we load in the new project on the [line](https://github.com/appleseedhq/appleseed/blob/1a0907c5dc3c491686cb2be4b3d4f9410965ba35/src/appleseed.studio/mainwindow/mainwindow.cpp#L245).

I hope I'm making myself clear.

As it stands now, loading in other projects (not just the built-in Cornell Box) in any order works fine, I've tested it myself. Is there a way you can pull the changes and see for yourself, or maybe I can record a video?

The main purpose of the snippet above is to bypass [this](https://github.com/appleseedhq/appleseed/blob/master/src/appleseed/foundation/utility/lazy.h#L369) line (`Lazy<renderer::TriangleTree>`)